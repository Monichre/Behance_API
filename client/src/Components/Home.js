import React, { Component } from 'react'
import Axios from 'axios'
import { Constants } from '../constants.js'

export default class Home extends Component {
    constructor(props) {
        super(props)

        this.state = {
            searchActive: false,
            searchTerm: ''
        }
    }
    componentWillMount() {
        Axios.get('/gallery').then((response) => {
            console.log(response)
        }).catch((error) => {
            console.log(error)
        })
    }
    runSearch(e) {
        e.preventDefault()
        this.setState({
            searchActive: true
        })
    }
    closeSearch(e) {
        e.preventDefault()
        this.setState({
            searchActive: false
        })
    }
    handleSearchInput(e) {
        e.preventDefault()
    }

    render() {
        return (
            <div id="Home">
                <header>
                    <i className="fa fa-bars"></i>
                    <div className="menu">
                        <div className={`search-wrapper ${this.state.searchActive ? 'active' : ''}`}>
                            <div className="input-holder">
                                <input type="text" className="search-input" value={this.state.searchTerm} onChange={this.handleSearchInput.bind(this)} placeholder="Type to search" />
                                <button className="search-icon" onClick={this.runSearch.bind(this)}><span></span></button>
                            </div>
                            <span className="close" onClick={this.closeSearch.bind(this)}></span>
                        </div>
                    </div>
                </header>

                <aside>
                    <h1>Dealer Inspire + Behance</h1>
                    <ul>

                    </ul>
                </aside>

                <div className="dashboard">
                    <canvas id="Chart"></canvas>
                    <div className="row">
                    </div>
                </div>
            </div>
        )
    }

}