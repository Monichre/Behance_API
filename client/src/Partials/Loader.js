import React from 'react'
export const Loader = () => (
    <div className="loader_bg">
        <div className="loader">
            <div className="loader__figure"></div>
            <p className="loader__label">Loading...</p>
        </div>
    </div>
)