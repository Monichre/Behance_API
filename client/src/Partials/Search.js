export const Search = (props) => (
    <div className='search-content'>
        <div className='results-column'>
            {
                this.props.searchResults.map((result) => (
                    <div className='searchResult'>
                        <h1>search result</h1>
                    </div>
                ))
            }
        </div>
    </div>
)