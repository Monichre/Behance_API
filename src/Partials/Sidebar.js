export const SideBar = () => (
    <div class="dashboard-sidebar">
        <div class="dashboard-sidebar__brand">

        </div>
        <div class="dashboard-nav">
        <ul>
            <li class="dashboard-nav__item"><a href="home"></a></li>
            <li class="dashboard-nav__item dashboard-nav__item--selected"></li>
            <li class="dashboard-nav__item"><a href="discover_items"></a></li>
            <li class="dashboard-nav__item"><a href="notifications"></a></li>
            <li class="dashboard-nav__item"><a href="settings"></a></li>
        </ul>
        </div>
</div>
)