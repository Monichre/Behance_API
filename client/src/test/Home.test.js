import React from 'react'
import sinon from 'sinon'
import Home from '../Components/Home'
import {Search} from '../Partials/Search'
import runSearch from './__mocks__/search-api.js'
import { shallowToJson } from 'enzyme-to-json';


describe('<Home />', () => {
    it('renders without crashing', () => {
        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        expect(shallowToJson(home_wrapper)).toMatchSnapshot()
    })
    it('should render a search icon button', () => {
        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        expect(home_wrapper.find('.search-icon')).toHaveLength(1)
    })
    it('should call handleClick() when clicked', () => {
        const spy = sinon.spy(Home.prototype, 'startSearch')
        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        home_wrapper.find('.search-icon').simulate('click')
        expect(spy.calledOnce).toEqual(true)
        expect(home_wrapper.find('.search-wrapper.active')).toHaveLength(1)
    })
    it('closes the search when button is clicked', async () => {
        
        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        home_wrapper.find('.search-icon').simulate('click')
        
        expect(home_wrapper.find('.search-wrapper.active')).toHaveLength(1)
        expect(home_wrapper.find('.search-wrapper.active .close')).toHaveLength(1)
        
        home_wrapper.find('.search-wrapper.active .close').simulate('click')
        expect(shallowToJson(home_wrapper)).toMatchSnapshot()

    });
    it('should update search term onChange', () => {
        const value = 'Liam'
        const onChange = jest.fn();
        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        const searchInput = home_wrapper.find('.search-input')
        const wrapper = shallow(<searchInput onChange={onChange} />)

        wrapper.simulate('change', value)
        expect(onChange).toBeCalledWith(value);
    })
    it('run the search on keypress enter', async () => {
        const handleKeyPress = jest.fn()
        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        const searchInput = home_wrapper.find('.search-input')
        const search_wrapper = shallow(<searchInput onKeyDown={handleKeyPress} />)

        search_wrapper.simulate('keyDown')
        expect(handleKeyPress).toHaveBeenCalledTimes(1);
    })
    it('expects handle keypress to return runSearch', async () => {
        const handleKeyPress = jest.fn()
        const runSearch = jest.fn()
        handleKeyPress.mockReturnValue(runSearch())

        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        const searchInput = home_wrapper.find('.search-input')
        const search_wrapper = shallow(<searchInput onKeyDown={handleKeyPress} />)

        search_wrapper.simulate('keyDown')
        expect(handleKeyPress()).toEqual(runSearch())
    })

    it('sets component state on runSearch', async () => {
        jest.mock('./__mocks__/search-api.js')

        const spy = sinon.spy(Home.prototype, 'runSearch')
        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        const data = await runSearch()
        const mock = jest.fn().mockImplementation(() => {
            home_wrapper.setState({
                searchResults: data
            })
        })
        const handleKeyPress = jest.fn()
        handleKeyPress.mockReturnValue(mock())
        home_wrapper.setState({
            searchTerm: 'Liam'
        })

        home_wrapper.simulate('runSearch')
        expect(home_wrapper.state().searchTerm).toBe('Liam')
        expect(shallowToJson(home_wrapper)).toMatchSnapshot()

        home_wrapper.simulate('handleKeyPress')
        expect(home_wrapper.state().searchResults).toEqual(data)

    });
    it('displays search results', async () => {
        jest.mock('./__mocks__/search-api.js')

        // const spy = sinon.spy(Home.prototype, 'runSearch')
        const home_wrapper = shallow(<Home fields={[]} items={[]} />)
        const data = await runSearch()
        const mock = jest.fn().mockImplementation(() => {
            home_wrapper.setState({
                searchResults: data
            })
        })
        const handleKeyPress = jest.fn()
        handleKeyPress.mockReturnValue(mock())
        home_wrapper.setState({
            searchTerm: 'Liam'
        })

        home_wrapper.simulate('runSearch')
        home_wrapper.simulate('handleKeyPress')
        
        const search_wrapper = mount(<Search searchResults={data}/>)
        expect(shallowToJson(search_wrapper)).toMatchSnapshot()

    });


});

describe('<Search />', () => {
    it('search results link to proper page', async () => {
        jest.mock('./__mocks__/search-api.js')

        const data = await runSearch()
        const search_wrapper = mount(<Search searchResults={data}/>)
        expect(search_wrapper.find('.searchResult a')).toBeTruthy()

    });
});