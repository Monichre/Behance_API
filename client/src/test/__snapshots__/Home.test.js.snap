// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> it sets component state on runSearch 1`] = `
<Home
  fields={Array []}
  items={Array []}
>
  <div
    id="Home"
  >
    <header>
      <i
        className="fa fa-bars"
      />
      <div
        className="menu"
      >
        <div
          className="search-wrapper "
        >
          <div
            className="input-holder"
          >
            <input
              className="search-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Type to search"
              type="text"
              value=""
            />
            <button
              className="search-icon"
              onClick={[Function]}
            >
              <span />
            </button>
          </div>
          <span
            className="close"
            onClick={[Function]}
          />
        </div>
      </div>
    </header>
    <aside>
      <h1>
        Dealer Inspire + Behance
      </h1>
      <ul />
    </aside>
    <div
      className="dashboard"
    >
      <Search
        searchResults={Array []}
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="search-content"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <h1>
            Search Results
          </h1>
          <div
            className="results-column"
          />
        </div>
      </Search>
      <CardGroup>
        <div
          className="ui cards"
        />
      </CardGroup>
    </div>
  </div>
</Home>
`;

exports[`<Home /> renders without crashing 1`] = `
<div
  id="Home"
>
  <header>
    <i
      className="fa fa-bars"
    />
    <div
      className="menu"
    >
      <div
        className="search-wrapper "
      >
        <div
          className="input-holder"
        >
          <input
            className="search-input"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Type to search"
            type="text"
            value=""
          />
          <button
            className="search-icon"
            onClick={[Function]}
          >
            <span />
          </button>
        </div>
        <span
          className="close"
          onClick={[Function]}
        />
      </div>
    </div>
  </header>
  <aside>
    <h1>
      Dealer Inspire + Behance
    </h1>
    <ul />
  </aside>
  <div
    className="dashboard"
  >
    <Search
      searchResults={Array []}
      style={
        Object {
          "display": "none",
        }
      }
    />
    <CardGroup />
  </div>
</div>
`;

exports[`<Home /> sets component state on runSearch 1`] = `
<div
  id="Home"
>
  <header>
    <i
      className="fa fa-bars"
    />
    <div
      className="menu"
    >
      <div
        className="search-wrapper "
      >
        <div
          className="input-holder"
        >
          <input
            className="search-input"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Type to search"
            type="text"
            value="Liam"
          />
          <button
            className="search-icon"
            onClick={[Function]}
          >
            <span />
          </button>
        </div>
        <span
          className="close"
          onClick={[Function]}
        />
      </div>
    </div>
  </header>
  <aside>
    <h1>
      Dealer Inspire + Behance
    </h1>
    <ul />
  </aside>
  <div
    className="dashboard"
  >
    <Search
      searchResults={
        Array [
          Object {
            "city": "London",
            "company": "",
            "country": "United Kingdom",
            "created_on": 1353324132,
            "display_name": "Liam Pitchford",
            "fields": Array [
              "Art Direction",
              "Motion Graphics",
              "Digital Art",
            ],
            "first_name": "Liam",
            "has_default_image": 0,
            "id": 1853107,
            "images": Object {
              "100": "https://mir-s3-cdn-cf.behance.net/user/100/4609491853107.58b2df5a7fef4.jpg",
              "115": "https://mir-s3-cdn-cf.behance.net/user/115/4609491853107.58b2df5a7fef4.jpg",
              "138": "https://mir-s3-cdn-cf.behance.net/user/138/4609491853107.58b2df5a7fef4.jpg",
              "230": "https://mir-s3-cdn-cf.behance.net/user/230/4609491853107.58b2df5a7fef4.jpg",
              "276": "https://mir-s3-cdn-cf.behance.net/user/276/4609491853107.58b2df5a7fef4.jpg",
              "50": "https://mir-s3-cdn-cf.behance.net/user/50/4609491853107.58b2df5a7fef4.jpg",
            },
            "last_name": "Pitchford",
            "location": "London, United Kingdom",
            "occupation": "Digital Artist",
            "state": "",
            "stats": Object {
              "appreciations": 4954,
              "comments": 47,
              "followers": 1251,
              "following": 303,
              "views": 47030,
            },
            "url": "https://www.behance.net/LiamPitchford",
            "username": "LiamPitchford",
            "website": "http://www.liampitchford.com",
          },
          Object {
            "city": "Munich",
            "company": "philipp rupprecht photography",
            "country": "Germany",
            "created_on": 1346001173,
            "display_name": "Philipp Rupprecht",
            "fields": Array [
              "Photography",
              "Digital Photography",
              "Retouching",
            ],
            "first_name": "Philipp",
            "has_default_image": 0,
            "id": 1519755,
            "images": Object {
              "100": "https://mir-s3-cdn-cf.behance.net/user/100/1519755.53b4e894e99dd.jpg",
              "115": "https://mir-s3-cdn-cf.behance.net/user/115/1519755.53b4e894e99dd.jpg",
              "138": "https://mir-s3-cdn-cf.behance.net/user/138/1519755.53b4e894e99dd.jpg",
              "230": "https://mir-s3-cdn-cf.behance.net/user/230/1519755.53b4e894e99dd.jpg",
              "276": "https://mir-s3-cdn-cf.behance.net/user/276/1519755.53b4e894e99dd.jpg",
              "50": "https://mir-s3-cdn-cf.behance.net/user/50/1519755.53b4e894e99dd.jpg",
            },
            "last_name": "Rupprecht",
            "location": "Munich, Germany",
            "occupation": "",
            "state": "",
            "stats": Object {
              "appreciations": 9654,
              "comments": 30,
              "followers": 4150,
              "following": 170,
              "views": 105391,
            },
            "url": "https://www.behance.net/philipprupprecht",
            "username": "philipprupprecht",
            "website": "http://www.philipprupprecht.de",
          },
          Object {
            "city": "",
            "company": "",
            "country": "",
            "created_on": 0,
            "display_name": "[Invalid User]",
            "fields": Array [],
            "first_name": "[Invalid User]",
            "has_default_image": 1,
            "id": 75433,
            "images": Object {
              "100": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-100.jpg?cb=264615658",
              "115": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-115.jpg?cb=264615658",
              "138": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-138.jpg?cb=264615658",
              "230": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-230.jpg?cb=264615658",
              "276": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-276.jpg?cb=264615658",
              "50": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-50.jpg?cb=264615658",
            },
            "last_name": "",
            "location": "",
            "occupation": "",
            "state": "",
            "stats": Object {
              "appreciations": 0,
              "comments": 89,
              "followers": 0,
              "following": 0,
              "views": 0,
            },
            "url": "#",
            "username": "",
            "website": "#",
          },
          Object {
            "city": "Beijing",
            "company": "Endless bright ( EB );  Netease Beijing",
            "country": "China",
            "created_on": 1386319472,
            "display_name": "liam lee",
            "fields": Array [
              "Graphic Design",
              "Print Design",
              "Branding",
            ],
            "first_name": "liam",
            "has_default_image": 0,
            "id": 4360425,
            "images": Object {
              "100": "https://mir-s3-cdn-cf.behance.net/user/100/54a6274360425.550253a1c4353.jpg",
              "115": "https://mir-s3-cdn-cf.behance.net/user/115/54a6274360425.550253a1c4353.jpg",
              "138": "https://mir-s3-cdn-cf.behance.net/user/138/54a6274360425.550253a1c4353.jpg",
              "230": "https://mir-s3-cdn-cf.behance.net/user/230/54a6274360425.550253a1c4353.jpg",
              "276": "https://mir-s3-cdn-cf.behance.net/user/276/54a6274360425.550253a1c4353.jpg",
              "50": "https://mir-s3-cdn-cf.behance.net/user/50/54a6274360425.550253a1c4353.jpg",
            },
            "last_name": "lee",
            "location": "Beijing, China",
            "occupation": "Graphic design lovers、A father",
            "state": "",
            "stats": Object {
              "appreciations": 6378,
              "comments": 166,
              "followers": 3780,
              "following": 579,
              "views": 76749,
            },
            "url": "https://www.behance.net/liamsoso",
            "username": "liamsoso",
            "website": "",
          },
          Object {
            "city": "",
            "company": "",
            "country": "",
            "created_on": 1255122662,
            "display_name": "Liam Ferguson",
            "fields": Array [
              "Automotive Design",
              "Industrial Design",
              "Product Design",
            ],
            "first_name": "Liam",
            "has_default_image": 1,
            "id": 127522,
            "images": Object {
              "115": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-115.jpg?cb=264615658",
              "138": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-138.jpg?cb=264615658",
              "50": "https://a5.behance.net/2604992a5a6a3839cf279cf9beb3cd63d5ce5354/img/profile/no-image-50.jpg?cb=264615658",
            },
            "last_name": "Ferguson",
            "location": "",
            "occupation": "",
            "state": "",
            "stats": Object {
              "appreciations": 4639,
              "comments": 5,
              "followers": 904,
              "following": 83,
              "views": 79091,
            },
            "url": "https://www.behance.net/LiamFerguson",
            "username": "LiamFerguson",
            "website": "",
          },
          Object {
            "city": "San Miguel de Tucumán",
            "company": "1.22/3",
            "country": "Argentina",
            "created_on": 1200316410,
            "display_name": "1.22/3 .",
            "fields": Array [
              "Art Direction",
              "Digital Art",
              "Graphic Design",
            ],
            "first_name": "1.22/3",
            "has_default_image": 0,
            "id": 57101,
            "images": Object {
              "100": "https://mir-s3-cdn-cf.behance.net/user/100/88440557101.5953d7ab6fbe5.jpg",
              "115": "https://mir-s3-cdn-cf.behance.net/user/115/88440557101.5953d7ab6fbe5.jpg",
              "138": "https://mir-s3-cdn-cf.behance.net/user/138/88440557101.5953d7ab6fbe5.jpg",
              "230": "https://mir-s3-cdn-cf.behance.net/user/230/88440557101.5953d7ab6fbe5.jpg",
              "276": "https://mir-s3-cdn-cf.behance.net/user/276/88440557101.5953d7ab6fbe5.jpg",
              "50": "https://mir-s3-cdn-cf.behance.net/user/50/88440557101.5953d7ab6fbe5.jpg",
            },
            "last_name": ".",
            "location": "San Miguel de Tucumán, Argentina",
            "occupation": "",
            "state": "",
            "stats": Object {
              "appreciations": 2061,
              "comments": 86,
              "followers": 461,
              "following": 445,
              "views": 34369,
            },
            "url": "https://www.behance.net/Qoya",
            "username": "Qoya",
            "website": "www.122.net",
          },
        ]
      }
      style={
        Object {
          "display": "none",
        }
      }
    />
    <CardGroup />
  </div>
</div>
`;
